web_app/tests/test_positions.py

```diff
@@ -0,0 +1,282 @@
+
+
+# Test cases for /api/create_position_with_transaction_data
+@pytest.mark.anyio
+async def test_create_position_success(client: AsyncClient) -> None:
+    """
+    Test for successfully creating a position with valid form data.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    form_data = {
+        "wallet_id": "valid_wallet_id",
+        "token_symbol": "ETH",
+        "amount": "1000",
+        "multiplier": 2,
+    }
+    mock_position = Mock()
+    mock_position.id = 123
+    mock_deposit_data = {
+        "caller": "mock_caller",
+        "pool_price": 100,
+        "pool_key": {
+            "token0": "mock_token0",
+            "token1": "mock_token1",
+            "fee": "mock_fee",
+            "tick_spacing": "mock_tick_spacing",
+            "extension": "mock_extension",
+        },
+        "deposit_data": {
+            "token": "mock_token",
+            "amount": "mock_amount",
+            "multiplier": "mock_multiplier",
+        },
+        "contract_address": "mock_contract_address",
+        "position_id": "123",
+    }
+
+    with (
+        patch(
+            "web_app.db.crud.PositionDBConnector.create_position"
+        ) as mock_create_position,
+        patch(
+            "web_app.contract_tools.mixins.deposit.DepositMixin.get_transaction_data"
+        ) as mock_get_transaction_data,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_contract_address_by_wallet_id"
+        ) as mock_get_contract_address,
+    ):
+        mock_create_position.return_value = mock_position
+        mock_get_transaction_data.return_value = mock_deposit_data
+        mock_get_contract_address.return_value = "mock_contract_address"
+        response = client.post("/api/create-position", json=form_data)
+        assert response.status_code == 200
+        print(response.json())
+        expected_response = {
+            "contract_address": "mock_contract_address",
+            "position_id": "123",
+            "caller": "mock_caller",
+            "pool_price": 100,
+            "pool_key": {
+                "token0": "mock_token0",
+                "token1": "mock_token1",
+                "fee": "mock_fee",
+                "tick_spacing": "mock_tick_spacing",
+                "extension": "mock_extension",
+            },
+            "deposit_data": {
+                "token": "mock_token",
+                "amount": "mock_amount",
+                "multiplier": "mock_multiplier",
+            },
+        }
+        assert response.json() == expected_response
+
+
+@pytest.mark.anyio
+async def test_create_position_invalid(client: AsyncClient) -> None:
+    """
+    Test for attempting to create a position with invalid input data, which should return a 422 error.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    response = client.post(
```

Comment:
add more edge cases, token_symbol is incorrect or amount is `str` type not `int`, add test cases, where `mulitpiler` is `str` instead of `int`, or None
You should try insert for `int` type params -> random `str` or None